haematite_ser/src/serialize_map.rs

```diff
@@ -0,0 +1,41 @@
+use std::collections::HashMap;
+
+use crate::error::{Error, Result};
+use crate::map_key::MapKeySerializer;
+use crate::wrap::{SerializeWrap, WrapType};
+use crate::Serializer;
+
+use serde::ser::Serialize;
+
+#[derive(Default)]
+pub struct SerializeMap {
+    map: HashMap<String, SerializeWrap>,
+    key: Option<String>,
+}
+
+impl serde::ser::SerializeMap for SerializeMap {
+    type Ok = WrapType;
+    type Error = Error;
+
+    fn serialize_key<T>(&mut self, key: &T) -> Result<()>
+    where
+        T: ?Sized + Serialize,
+    {
+        self.key = key.serialize(MapKeySerializer {}).ok();
```

Comment:
So if an error occurs here (like `KeyMustBeString`), it simply causes that key/value pair to be dropped?